### PR TITLE
Support only current Nodejs LTS versions

### PR DESCRIPTION
**Pull request by orien**
### Context

Node.js v12, v14, and v16 reached their [end of life] on 2022-04-30, 2023-04-30, and 2023-09-11, respectively.

To support new features introduced in modern versions of Node.js, I propose removing support for these old, unmaintained versions of Node.js.

[end of life]: https://github.com/nodejs/Release#end-of-life-releases

### Change

Bump the minimum Node.js version to 18.0.0.

### CODE REVIEW

**Comment by orien** [comment]
I signed the CLA, but there was some error: when I attempt to sign again it says:

image

File: package.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
     "colors": "1.4.0"
   },
   "engines": {
-    "node": ">=12.0.0"
+    "node": ">=16.0.0"
```
Comment (orien) [inline package.json:65]: Bump the minimum Node.js version to 16.0.0.

File: bin/dev [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 const oclif = require('@oclif/core')
 
-const path = require('path')
+const path = require('node:path')
```
Comment (orien) [inline bin/dev:5]: Minor refactoring: prefix requires for Node.js modules with `node:`.

File: src/tarballs/node.ts [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
-import * as path from 'path'
+import * as path from 'node:path'
 import * as fs from 'fs-extra'
-import {pipeline as pipelineSync} from 'node:stream'
+import {pipeline} from 'node:stream/promises'
 import {log} from '../log'
 import {exec as execSync} from 'node:child_process'
 import {promisify} from 'node:util'
 import got from 'got'
-const pipeline = promisify(pipelineSync)
```
Comment (orien) [inline src/tarballs/node.ts:9]: Use the  promisified `pipeline` function provided by Node.js. This was introduced in Node.js 15.

**Comment by orien** [comment]
Hi @mdonnalley, I'm wondering if you could please review my first PR on this project?

**Comment by orien** [comment]
That makes sense. Thanks @mdonnalley.

I notice that Node.js v16 will reach its [end of life on September 11](https://github.com/nodejs/release#release-schedule). Do you plan to drop support for that in the major release? Shall I update this PR to make v18 the minimum?

**Comment by mdonnalley** [comment]
@orien Yes, we'll go to node 18.

**Comment by mdonnalley** [comment]
@orien Thanks for the contribution. This will be going into the `prerelease/v4` branch which will be released as a new major version under a `beta` tag.

**Pull request merged**